.github/workflows/rust.yml

```diff
@@ -63,8 +63,8 @@ jobs:
       - run: rustup update
       - run: rustup target add ${{ matrix.target }}
       # TODO: we should probably manage the container in the test itself
-      # TODO: do a proper functional check on the container startup instead of waiting 15 seconds
       - run: docker run -d -p 8000:8000 stellar/quickstart:soroban-dev --standalone --enable-soroban-rpc --enable-core-artificially-accelerate-time-for-testing --protocol-version 20 && sleep 15
+      - run: while ! [ "$(curl -s --fail localhost:8000 | jq '.history_latest_ledger')" -gt 0 ]; do echo waiting; sleep 1; done
```

Comment:
you may want to merge this with the prior command

Reply:
I'd rather keep it separate so that it is clear in the GitHub Action when the container has started vs the build is waiting for the container to start. That seemed to work well here: 
https://github.com/stellar/starlight/blob/919096552e44eec5ec3fd6278922396926239874/.github/workflows/sdk.yml#L59-L62

Reply:
No strong opinions, I think it's easier to read if done in the same step (spawn the docker container and make sure it's working).